third_party/move/move-compiler-v2/src/env_pipeline/inliner.rs

```diff
@@ -436,6 +441,50 @@ impl<'env> Inliner<'env> {
     }
 }
 
+/// Expression rewriter for lambda spec
+struct LambdaSpecRewriter<'env> {
+    env: &'env GlobalEnv,
+    result_flag: bool,
+}
+
+impl<'env> LambdaSpecRewriter<'env> {
+    fn new(env: &'env GlobalEnv, result_flag: bool) -> Self {
+        LambdaSpecRewriter { env, result_flag }
+    }
+}
+
+impl<'env> ExpRewriterFunctions for LambdaSpecRewriter<'env> {
```

Comment:
I guess we are not longer doing it like this with the new discussed design?

Reply:
Yeah, once we synthesize the spec block into loop invariants, I don't think we need to insert spec block before and after the inlined lambda. I will go back and fix this later.